src/sparse/node.rs

```diff
@@ -325,6 +325,18 @@ impl Node {
     }
 }
 
+impl TryFrom<Buffer> for Node {
+    type Error = DeserializeError;
+
+    fn try_from(value: Buffer) -> Result<Self, Self::Error> {
+        let node = Self { buffer: value };
+
+        let _prefix: Prefix = node.bytes_prefix()[0].try_into()?;
```

Comment:
I think you need check the prefix here

Reply:
Yes, that is actually what this line is supposed to do: We look at the byte from the buffer at the prefix location and try to convert it into a `Prefix` using the `Prefix`'s [TryFrom impl](https://github.com/FuelLabs/fuel-merkle/pull/131/files#diff-ead3431f00f7caf7fb4ee2867548b8044c4f63fe0343de509fe941b86bbcea17R28). This checks that the byte is either an internal `Node` (0x01) or a `Leaf` (0x00); if it is not one of those, it will return a `DeserializeError`, and this `try_from` will propagate the `DeserializeError` upwards.